connectors/aws/aws-s3/src/main/java/io/camunda/connector/aws/s3/core/S3Executor.java

```diff
@@ -0,0 +1,145 @@
+            .region(Region.of(s3Request.getConfiguration().region()))
+            .build(),
+        createDocument);
+  }
+
+  public Object execute(S3Action s3Action) {
+    return switch (s3Action) {
+      case DeleteObject deleteObject -> delete(deleteObject);
+      case DownloadObject downloadObject -> download(downloadObject);
+      case UploadObject uploadObject -> upload(uploadObject);
+    };
+  }
+
+  private Object upload(UploadObject uploadObject) {
+    Long contentLength = uploadObject.document().metadata().getSize();
+    String contentType = uploadObject.document().metadata().getContentType();
+
+    PutObjectRequest putObjectRequest =
+        PutObjectRequest.builder()
+            .bucket(uploadObject.bucket())
+            .key(
+                Optional.ofNullable(uploadObject.key())
+                    .orElse(uploadObject.document().metadata().getFileName()))
+            .contentLength(contentLength)
+            .contentType(contentType)
+            .build();
+
+    this.s3Client.putObject(
+        putObjectRequest,
+        RequestBody.fromInputStream(uploadObject.document().asInputStream(), contentLength));
+
+    return new UploadResponse(
+        uploadObject.bucket(),
+        uploadObject.key(),
+        String.format("https://%s.s3.amazonaws.com/%s", uploadObject.bucket(), uploadObject.key()));
+  }
+
+  private DownloadResponse download(DownloadObject downloadObject) {
+    GetObjectRequest getObjectRequest =
+        GetObjectRequest.builder()
+            .bucket(downloadObject.bucket())
+            .key(downloadObject.key())
+            .build();
+
+    ResponseInputStream<GetObjectResponse> getObjectResponse =
+        this.s3Client.getObject(getObjectRequest);
+
+    if (!downloadObject.asFile()) {
+      try {
+        return retrieveResponseWithContent(
+            downloadObject.bucket(), downloadObject.key(), getObjectResponse);
+      } catch (IOException e) {
+        log.error("An error occurred while trying to read and parse the downloaded file", e);
+        throw new RuntimeException(e);
+      }
+    } else {
+      return this.createDocument
+          .andThen(
+              document ->
+                  new DownloadResponse(
+                      downloadObject.bucket(),
+                      downloadObject.key(),
+                      new Element.DocumentContent(document)))
+          .apply(
+              DocumentCreationRequest.from(getObjectResponse)
+                  .contentType(getObjectResponse.response().contentType())
+                  .fileName(downloadObject.key())
+                  .build());
+    }
+  }
+
+  private DownloadResponse retrieveResponseWithContent(
+      String bucket, String key, ResponseInputStream<GetObjectResponse> responseResponseInputStream)
+      throws IOException {
+    byte[] rawBytes = responseResponseInputStream.readAllBytes();
+    return switch (responseResponseInputStream.response().contentType()) {
```

Comment:
not sure here, but be aware that sometimes the case is weird (uppercase letters etc), and also we might have `application/json; charset=utf-8`. Maybe it's not applicable here though.

Reply:
Maybe should we use [MimeType](https://docs.oracle.com/javase/8/docs/api///?javax/activation/MimeType.html) class for this

Reply:
The ContentType returned should always be standard
![image](https://github.com/user-attachments/assets/cbbdbae2-cc80-400a-ad43-5f6f6c963d8b)